lib/cli/config.js

```diff
@@ -28,14 +28,34 @@ exports.CONFIG_FILES = [
   '.mocharc.json'
 ];
 
+const isModuleNotFoundError = err =>
+  err.code !== 'MODULE_NOT_FOUND' ||
+  err.message.indexOf('Cannot find module') !== -1;
+
 /**
- * Parsers for various config filetypes.  Each accepts a filepath and
+ * Parsers for various config filetypes. Each accepts a filepath and
  * returns an object (but could throw)
  */
 const parsers = (exports.parsers = {
   yaml: filepath =>
     require('js-yaml').safeLoad(fs.readFileSync(filepath, 'utf8')),
-  js: filepath => require(filepath),
+  js: filepath => {
+    const cwdFilepath = path.resolve(filepath);
+    try {
+      debug(`parsers: load using cwd-relative path: "${cwdFilepath}"`);
+      return require(cwdFilepath);
+    } catch (err) {
+      if (
+        isModuleNotFoundError(err) &&
+        require('validate-npm-package-name')(filepath).validForNewPackages
```

Comment:
what's the rationale for this restriction?  `validate-npm-package-name` seems unnecessarily strict.  just `require(filepath)` if the resolved path doesn't work.

Reply:
FWIW this also looks like a breaking change.

Reply:
It _was_ to replace `module.require` loading of files with packages instead. But that's DOA now.